extension.js

```diff
@@ -168,6 +179,48 @@ const ExtensionLayout = new Lang.Class({
     GLib.spawn_command_line_async(cmd);
   },
 
+  _findNewStreamerEntries:function(lastList, currentList, detectGameChange) {
+    detectGameChange = detectGameChange || false;
+    if ( lastList.length == 0 )
+      return currentList;
+
+    let newEntries = [];
+    // filter out all lastList items from currentList
+    currentList.forEach(entry => {
+      let match = false;
+
+      // Use a for loop so we can break it off when we match
+      for ( var i = 0; i < lastList.length; i++ ) {
+        let last = lastList[i];
+        if ( entry.streamer == last.streamer )
+          if ( !detectGameChange || ( detectGameChange && entry.game == last.game ) ) {
+            match = true;
```

Comment:
Wouldn't it be easiert to use a map? `{}[Streamer] =` ...

Reply:
I don't have a way to answer that question or provide reasoning for this method because I don't spend much time in javascript.

Do you mean [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) or [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)?

Reply:
This version breaks after a while, it seems. Looks like it happens when new streamers show up. Non-events and game changes don't seem to be an issue.